File: AspNetMvcUnleashed/0672329980_source/Chapter19/VB/UnleashedBlog/UnleashedBlog.Tests/Models/FakeBlogRepository.vb
Imports System.Collections.Generic
Imports System.Linq

Public Class FakeBlogRepository
    Inherits BlogRepositoryBase

    Private _blogEntries As New List(Of BlogEntry)

    Protected Overrides Function QueryBlogEntries() As IQueryable(Of BlogEntry)
        Return _blogEntries.AsQueryable
    End Function


    Public Overrides Function ListBlogEntries() As List(Of BlogEntry)
        Return Me.QueryBlogEntries().ToList()
    End Function

    Public Overrides Sub CreateBlogEntry(ByVal blogEntryToCreate As BlogEntry)
        _blogEntries.Add(blogEntryToCreate)
    End Sub
End Class

File: AspNetMvcUnleashed/0672329980_source/Chapter19/CS/UnleashedBlog/UnleashedBlog.Tests/Models/FakeBlogRepository.cs
using System.Collections.Generic;
using System.Linq;
using UnleashedBlog.Models;

namespace UnleashedBlog.Tests.Models
{
    public class FakeBlogRepository : BlogRepositoryBase
    {
        private List<BlogEntry> _blogEntries = new List<BlogEntry>();

        protected override IQueryable<BlogEntry> QueryBlogEntries()
        {
            return _blogEntries.AsQueryable();
        }

        public override List<BlogEntry> ListBlogEntries()
        {
            return this.QueryBlogEntries().ToList();
        }

        public override void CreateBlogEntry(BlogEntry blogEntryToCreate)
        {
            _blogEntries.Add(blogEntryToCreate);
        }
    }
}
